app/src/main/java/ch/hikemate/app/model/authentication/AuthViewModel.kt

```diff
@@ -47,18 +63,26 @@ class AuthViewModel(
       context: Context,
       startAddAccountIntentLauncher: ManagedActivityResultLauncher<Intent, ActivityResult>?,
   ) {
+    _loading.value = true
 
     repository.signInWithGoogle(
         onSuccess = { user: FirebaseUser? ->
           profileRepository.createProfile(
-              // TODO handle errors
               user,
-              onSuccess = { _currentUser.value = user },
-              onFailure = {},
+              onSuccess = {
+                _currentUser.value = user
+                _loading.value = true
```

Comment:
Warning: This should not be set to true, here, we stop loading

Reply:
Good catch!